project/paths.ts

```diff
@@ -4,32 +4,52 @@ import process from "process"
 import { showLoadingMessage } from "../main"
 import { defaultTsconfig } from "../generate_library_defs/generate_tsconfig"
 import { ParsedArgs } from "../parse_args"
+import type { Matcher } from "anymatch"
+import { allAssetExtensions } from "./assets"
 
 // TODO: Do sourceTsPath and destGdPath have to be relative?
 
-export class Paths {
-  /** Where the .ts files live, e.g. ./src */
+export class Paths implements PathProvider {
   sourceTsPath: string
-
-  /** Where the compiled .gd files go, e.g. ./compiled */
   destGdPath: string
-
-  /** The root path of the project */
   rootPath: string
-
-  /** The full path to the tsconfig file. e.g. /Users/johnfn/GodotProject/tsconfig.json */
   tsconfigPath: string
-
-  /** The path to the Godot definitions folder for unchanging library definitions.
-   * e.g. /Users/johnfn/GodotProject/_godot_defs/static */
   staticGodotDefsPath: string
-
-  /** The path to the Godot definitions folder for definitions based off user files.
-   * e.g. /Users/johnfn/GodotProject/_godot_defs/dynamic */
   dynamicGodotDefsPath: string
-
-  /** The path to the Godot repository, e.g. /Users/johnfn/Godot */
   godotSourceRepoPath: string | undefined
+  csgClassesPath: string
+  websocketClassesPath: string
+  normalClassesPath: string
+  gdscriptPath: string
+
+  additionalIgnores: string[]
+  tsFileIgnores: string[]
+
+  resPathToFsPath(resPath: string) {
+    return path.join(this.rootPath, resPath.slice("res://".length))
+  }
+
+  fsPathToResPath(fsPath: string) {
+    return "res://" + fsPath.slice(this.rootPath.length + 1)
+  }
+
+  ignoredPaths(): Matcher {
+    return [
+      "**/node_modules/**",
+      "**/_godot_defs/**",
+      "**/.git/**",
+      ...this.additionalIgnores,
```

Comment:
Ergh... I don't really like how we have to do `"**/node_modules/**"` because tsconfig also has file ignores, but it's just `"./node_modules/**"` in tsconfig, so it's a little inconsistent. 

I assume you did `**/blah/**` because choikar is giving us back absolute paths? Is there a way to standardize the paths chokidar gives us back so that node_modules is just ./node_modules?

Reply:
The **/node_modules/** is also helping in the cases in which there is some other node_modules folder somewhere.
But you are right. Maybe there is some way to check if TS would ignore a file or not and use that. Will have a look.